### PR TITLE
Add Last Oasis Egg

### PR DESCRIPTION
**Pull request by CrazyDev05**
# Description

This PR aims to add a Last Oasis Dedicated Server Egg.

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
6. [x] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/steamcmd_servers/lastoasis/egg-last-oasis.json [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "Server Slots",
+            "description": "Limited to 100",
+            "env_variable": "SLOTS",
+            "default_value": "5",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string|max:100",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/lastoasis/egg-last-oasis.json:71]: make this required to be a number

**Comment by gOOvER** [comment]
looks Like my Egg ;)

File: game_eggs/steamcmd_servers/lastoasis/egg-last-oasis.json [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "[Advanced] Branch Name",
+            "description": "",
+            "env_variable": "SRCDS_BETAID",
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "string|nullable|in:public,classic",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/lastoasis/egg-last-oasis.json:121]: you eather set the default to public or just remove the validation or do `string|in: ,classic`

File: game_eggs/steamcmd_servers/lastoasis/egg-last-oasis.json [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            "env_variable": "SLOTS",
+            "default_value": "5",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string|max:100",
+            "field_type": "text"
+        },
+        {
+            "name": "Provider Key",
+            "description": "You will need to login to \r\n\r\nhttps:\/\/myrealm.lastoasis.gg\/ \r\n\r\nusing Steam, and grab a few keys used for the server setup. Specifically, you will require a CustomerKey, which is used in all servers you\u2019re using on that realm, and a ProviderKey.\r\n\r\nProviderKey can be utilized as many times as you want in as many different servers in the same realm. CustomerKey must be the same across all servers in a realm.",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/lastoasis/egg-last-oasis.json:76]: Please make this a litte shorter and mention it in the readme

File: game_eggs/steamcmd_servers/lastoasis/egg-last-oasis.json [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            "env_variable": "PROVIDER_KEY",
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string|max:50",
+            "field_type": "text"
+        },
+        {
+            "name": "Customer Key",
+            "description": "You will need to login to \r\n\r\nhttps:\/\/myrealm.lastoasis.gg\/ \r\n\r\nusing Steam, and grab a few keys used for the server setup. Specifically, you will require a CustomerKey, which is used in all servers you\u2019re using on that realm, and a ProviderKey.\r\n\r\nProviderKey can be utilized as many times as you want in as many different servers in the same realm. CustomerKey must be the same across all servers in a realm.",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/lastoasis/egg-last-oasis.json:86]: Same here

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

**Comment by QuintenQVD0** [comment]
> looks Like my Egg ;)

If it is and he did not confirm with you he can share this then this will be closed ofc

**Comment by gOOvER** [comment]
you took my pr #1548 and only Set you AS dev?

sry, thats a nogo

**Comment by QuintenQVD0** [comment]
> you took my pr #1548 and only Set you AS dev?
> 
> sry, thats a nogo

ofc let me handle this

**Comment by QuintenQVD0** [comment]
@CrazyDev05 We do not tollerate you claiming someone else egg without there permission so this will be closed

**Pull request closed without merging**